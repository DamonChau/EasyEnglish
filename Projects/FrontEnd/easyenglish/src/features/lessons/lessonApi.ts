/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { api } from "../../services/api";
import { Lessons } from "../../models/types";
export type LessonsResponse = Lessons[];

export const lessonApi = api.injectEndpoints({
  endpoints: (build) => ({
    getAllLessons: build.query<LessonsResponse, void>({
      query: () => ({
        url: `api/Lesson/GetAllLessons/`,
      }),
      providesTags: (result = []) => [
        ...result.map(({ id }) => ({ type: "lessons", id } as const)),
        { type: "lessons" as const, id: "LIST" },
      ],
    }),
    getAllLessonsByType: build.query<LessonsResponse, string>({
        query: (lessonType) => ({
          url: `api/Lesson/GetAllLessonByType/${lessonType}`,
        }),
        providesTags: (result = []) => [
          ...result.map(({ id }) => ({ type: "lessons", id } as const)),
          { type: "lessons" as const, id: "LIST" },
        ],
      }),

    getLesson: build.query<Lessons, string>({
      query: (id) => `api/Lesson/Details/${id}`,
      providesTags: (result, error, arg) => [
        { type: "lessons", id: result?.id },
      ],
    }),
    addLesson: build.mutation<Lessons, Partial<Lessons>>({
      query: (body) => ({
        url: "api/Lesson/Create",
        method: "POST",
        body,
      }),
      invalidatesTags: [{ type: "lessons", id: "LIST" }],
    }),
    updateLesson: build.mutation<Lessons, Partial<Lessons>>({
      query(body) {
        return {
          url: `api/Lesson/Edit`,
          method: "PUT",
          body,
        };
      },
      invalidatesTags: (lesson) => [{ type: "lessons", id: lesson?.id }],
    }),
    deleteLesson: build.mutation<{ success: boolean; id: string }, string>({
      query(id) {
        return {
          url: `api/Lesson/Delete/${id}`,
          method: "DELETE",
        };
      },
      invalidatesTags: (lesson) => [{ type: "lessons", id: lesson?.id }],
    }),
  }),
});

export const {
    useAddLessonMutation,
    useDeleteLessonMutation,
    useGetAllLessonsByTypeQuery,
    useGetAllLessonsQuery,
    useGetLessonQuery,
    useUpdateLessonMutation
} = lessonApi;
