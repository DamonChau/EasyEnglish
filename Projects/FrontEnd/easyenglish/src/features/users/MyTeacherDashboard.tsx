/* eslint-disable @typescript-eslint/no-empty-interface */
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */
import React, { useEffect, useState } from "react";
import AvailableTeacher from "./AvailableTeacher";
import MyTeacher from "./MyTeacher";

const MyTeacherDashboard = () => {
  return (
    <div>
      <section
        className="hero-wrap hero-wrap-2"
        style={{ backgroundImage: 'url("images/bg_1.jpg")' }}
        data-stellar-background-ratio="0.5"
      >
        <div className="overlay" />
        <div className="container">
          <div className="row no-gutters slider-text align-items-center justify-content-center">
            <div className="col-md-9 text-center">
              <h1 className="mb-2 bread">Teacher Manager</h1>
              <p className="breadcrumbs">
                <span className="mr-2">
                  <a href="/">
                    Home <i className="ion-ios-arrow-forward" />
                  </a>
                </span>{" "}
                <span>
                  Teacher Manager
                  <i className="ion-ios-arrow-forward" />
                </span>
              </p>
            </div>
          </div>
        </div>
      </section>
      <section className="ftco-section">
        <div className="container">
          <div className="row">
            <div className="col-lg-10 bg-light">
              <div>
                <AvailableTeacher />
              </div>
            </div>
            <div className="col-lg-10 bg-light mt-5">
              <div>
                <MyTeacher />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  );
};

export default MyTeacherDashboard;
