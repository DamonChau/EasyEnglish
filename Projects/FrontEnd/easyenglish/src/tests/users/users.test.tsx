/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable no-debugger */
/* eslint-disable @typescript-eslint/no-unused-vars */
import "whatwg-fetch";
import React from "react";
import { waitFor, screen, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom";
import { renderWithProviders, store } from "../test-utils";
import MyTeacherDashboard from "../../features/users/MyTeacherDashboard";
import MyStudentDashboard from "../../features/users/MyStudentDashboard";
import NewUserAccount from "../../features/users/NewUserAccount";
import UserManager from "../../features/users/UserManager";
import { createStore } from "../../services/index";
import { generateMockUser } from "../users/mock";
import { mockLessonsData } from "../lessons/mock";

describe("Users form", () => {
  test("MyTeacherDashboard Snapshot", () => {
    const item = renderWithProviders(<MyTeacherDashboard />, {
      store: createStore({
        preloadedState: {
          auth: {
            token: null,
            user: generateMockUser(),
            isAuthenticated: false,
            refreshToken: null,
          },
        },
      }),
    });
    expect(item).toMatchSnapshot();
  });
  test("MyStudentDashboard Snapshot", () => {
    const item = renderWithProviders(<MyStudentDashboard />, {
      store: createStore({
        preloadedState: {
          auth: {
            token: null,
            user: generateMockUser(),
            isAuthenticated: false,
            refreshToken: null,
          },
        },
      }),
    });
    expect(item).toMatchSnapshot();
  });
  test("NewUserAccount Snapshot", () => {
    const item = renderWithProviders(<NewUserAccount />, {
      store: createStore({
        preloadedState: {
          auth: {
            token: null,
            user: generateMockUser(),
            isAuthenticated: false,
            refreshToken: null,
          },
        },
      }),
    });
    expect(item).toMatchSnapshot();
  });
  test("UserManager Snapshot", () => {
    const item = renderWithProviders(<UserManager />, {
      store: createStore({
        preloadedState: {
          auth: {
            token: null,
            user: generateMockUser(),
            isAuthenticated: false,
            refreshToken: null,
          },
        },
      }),
    });
    expect(item).toMatchSnapshot();
  });
});
