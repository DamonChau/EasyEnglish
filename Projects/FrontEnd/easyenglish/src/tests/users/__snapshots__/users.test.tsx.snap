// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users form MyStudentDashboard Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="hero-wrap hero-wrap-2"
          data-stellar-background-ratio="0.5"
          style="background-image: url(images/bg_1.jpg);"
        >
          <div
            class="overlay"
          />
          <div
            class="container"
          >
            <div
              class="row no-gutters slider-text align-items-center justify-content-center"
            >
              <div
                class="col-md-9 text-center"
              >
                <h1
                  class="mb-2 bread"
                >
                  Student Manager
                </h1>
                <p
                  class="breadcrumbs"
                >
                  <span
                    class="mr-2"
                  >
                    <a
                      href="/"
                    >
                      Home 
                      <i
                        class="ion-ios-arrow-forward"
                      />
                    </a>
                  </span>
                   
                  <span>
                    Student Manager
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </span>
                </p>
              </div>
            </div>
          </div>
        </section>
        <section
          class="ftco-section"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-lg-10 bg-light"
              >
                <div>
                  <p>
                    <em>
                      Loading...
                    </em>
                  </p>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-lg-10 bg-light mt-2"
              >
                <div
                  class="d-flex justify-content-center align-items-start m-2"
                >
                  <div
                    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSuccess MuiChip-clickable MuiChip-clickableColorSuccess MuiChip-filledSuccess mr-2 css-18ck578-MuiButtonBase-root-MuiChip-root"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      Assigned
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </div>
                  <div
                    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-nz3zgg-MuiButtonBase-root-MuiChip-root"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      Done
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </div>
                </div>
                <div>
                  <p>
                    <em>
                      Loading...
                    </em>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
    <script
      defer=""
      nonce="undefined"
      src="https://accounts.google.com/gsi/client"
    />
  </body>,
  "container": <div>
    <div>
      <section
        class="hero-wrap hero-wrap-2"
        data-stellar-background-ratio="0.5"
        style="background-image: url(images/bg_1.jpg);"
      >
        <div
          class="overlay"
        />
        <div
          class="container"
        >
          <div
            class="row no-gutters slider-text align-items-center justify-content-center"
          >
            <div
              class="col-md-9 text-center"
            >
              <h1
                class="mb-2 bread"
              >
                Student Manager
              </h1>
              <p
                class="breadcrumbs"
              >
                <span
                  class="mr-2"
                >
                  <a
                    href="/"
                  >
                    Home 
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </a>
                </span>
                 
                <span>
                  Student Manager
                  <i
                    class="ion-ios-arrow-forward"
                  />
                </span>
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="ftco-section"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-10 bg-light"
            >
              <div>
                <p>
                  <em>
                    Loading...
                  </em>
                </p>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-lg-10 bg-light mt-2"
            >
              <div
                class="d-flex justify-content-center align-items-start m-2"
              >
                <div
                  class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSuccess MuiChip-clickable MuiChip-clickableColorSuccess MuiChip-filledSuccess mr-2 css-18ck578-MuiButtonBase-root-MuiChip-root"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                  >
                    Assigned
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
                <div
                  class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-nz3zgg-MuiButtonBase-root-MuiChip-root"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                  >
                    Done
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </div>
              <div>
                <p>
                  <em>
                    Loading...
                  </em>
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Users form MyTeacherDashboard Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="hero-wrap hero-wrap-2"
          data-stellar-background-ratio="0.5"
          style="background-image: url(images/bg_1.jpg);"
        >
          <div
            class="overlay"
          />
          <div
            class="container"
          >
            <div
              class="row no-gutters slider-text align-items-center justify-content-center"
            >
              <div
                class="col-md-9 text-center"
              >
                <h1
                  class="mb-2 bread"
                >
                  Teacher Manager
                </h1>
                <p
                  class="breadcrumbs"
                >
                  <span
                    class="mr-2"
                  >
                    <a
                      href="/"
                    >
                      Home 
                      <i
                        class="ion-ios-arrow-forward"
                      />
                    </a>
                  </span>
                   
                  <span>
                    Teacher Manager
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </span>
                </p>
              </div>
            </div>
          </div>
        </section>
        <section
          class="ftco-section"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-lg-10 bg-light"
              >
                <div>
                  <p>
                    <em>
                      Loading...
                    </em>
                  </p>
                </div>
              </div>
              <div
                class="col-lg-10 bg-light mt-5"
              >
                <div>
                  <p>
                    <em>
                      Loading...
                    </em>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
    <script
      defer=""
      nonce="undefined"
      src="https://accounts.google.com/gsi/client"
    />
  </body>,
  "container": <div>
    <div>
      <section
        class="hero-wrap hero-wrap-2"
        data-stellar-background-ratio="0.5"
        style="background-image: url(images/bg_1.jpg);"
      >
        <div
          class="overlay"
        />
        <div
          class="container"
        >
          <div
            class="row no-gutters slider-text align-items-center justify-content-center"
          >
            <div
              class="col-md-9 text-center"
            >
              <h1
                class="mb-2 bread"
              >
                Teacher Manager
              </h1>
              <p
                class="breadcrumbs"
              >
                <span
                  class="mr-2"
                >
                  <a
                    href="/"
                  >
                    Home 
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </a>
                </span>
                 
                <span>
                  Teacher Manager
                  <i
                    class="ion-ios-arrow-forward"
                  />
                </span>
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="ftco-section"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-10 bg-light"
            >
              <div>
                <p>
                  <em>
                    Loading...
                  </em>
                </p>
              </div>
            </div>
            <div
              class="col-lg-10 bg-light mt-5"
            >
              <div>
                <p>
                  <em>
                    Loading...
                  </em>
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Users form NewUserAccount Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="hero-wrap hero-wrap-2"
          data-stellar-background-ratio="0.5"
          style="background-image: url(images/bg_1.jpg);"
        >
          <div
            class="overlay"
          />
          <div
            class="container"
          >
            <div
              class="row no-gutters slider-text align-items-center justify-content-center"
            >
              <div
                class="col-md-9 text-center"
              >
                <h1
                  class="mb-2 bread"
                >
                  Login
                </h1>
                <p
                  class="breadcrumbs"
                >
                  <span
                    class="mr-2"
                  >
                    <a
                      href="/"
                    >
                      Home 
                      <i
                        class="ion-ios-arrow-forward"
                      />
                    </a>
                  </span>
                   
                  <span>
                    New User Account 
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </span>
                </p>
              </div>
            </div>
          </div>
        </section>
        <section
          class="ftco-section ftco-no-pt ftco-no-pb contact-section"
        >
          <div
            class="container"
          >
            <div
              class="h-100 d-flex align-items-center justify-content-center flex-column"
            >
              <div
                class="col-md-6 p-4 p-md-6 order-md-last bg-light"
              >
                <form
                  action="#"
                >
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="username"
                    >
                      User Name (*)
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="username"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="email"
                    >
                      Email (*)
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="email"
                        type="email"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="password"
                    >
                      Password (*)
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="password"
                        type="password"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="retypePassword"
                    >
                      Retype Password (*)
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="retypePassword"
                        type="password"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="aliasName"
                    >
                      Alias Name (*)
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="aliasName"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="phoneno"
                    >
                      Phone No
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="phoneno"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="address"
                    >
                      Address
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="address"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="form-group row"
                  >
                    <label
                      class="col-sm-4 col-form-label"
                      for="billingAddress"
                    >
                      Billing Address
                    </label>
                    <div
                      class="col-sm-8"
                    >
                      <input
                        class="form-control"
                        name="billingAddress"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex flex-row-reverse"
                  >
                    <button
                      class="btn btn-primary py-2 px-3"
                      type="submit"
                    >
                      Submit
                    </button>
                    <button
                      class="btn btn-primary py-2 px-3 mr-2"
                      type="button"
                    >
                      Cancel
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
    <script
      defer=""
      nonce="undefined"
      src="https://accounts.google.com/gsi/client"
    />
  </body>,
  "container": <div>
    <div>
      <section
        class="hero-wrap hero-wrap-2"
        data-stellar-background-ratio="0.5"
        style="background-image: url(images/bg_1.jpg);"
      >
        <div
          class="overlay"
        />
        <div
          class="container"
        >
          <div
            class="row no-gutters slider-text align-items-center justify-content-center"
          >
            <div
              class="col-md-9 text-center"
            >
              <h1
                class="mb-2 bread"
              >
                Login
              </h1>
              <p
                class="breadcrumbs"
              >
                <span
                  class="mr-2"
                >
                  <a
                    href="/"
                  >
                    Home 
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </a>
                </span>
                 
                <span>
                  New User Account 
                  <i
                    class="ion-ios-arrow-forward"
                  />
                </span>
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="ftco-section ftco-no-pt ftco-no-pb contact-section"
      >
        <div
          class="container"
        >
          <div
            class="h-100 d-flex align-items-center justify-content-center flex-column"
          >
            <div
              class="col-md-6 p-4 p-md-6 order-md-last bg-light"
            >
              <form
                action="#"
              >
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="username"
                  >
                    User Name (*)
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="username"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="email"
                  >
                    Email (*)
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="email"
                      type="email"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="password"
                  >
                    Password (*)
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="password"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="retypePassword"
                  >
                    Retype Password (*)
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="retypePassword"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="aliasName"
                  >
                    Alias Name (*)
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="aliasName"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="phoneno"
                  >
                    Phone No
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="phoneno"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="address"
                  >
                    Address
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="address"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group row"
                >
                  <label
                    class="col-sm-4 col-form-label"
                    for="billingAddress"
                  >
                    Billing Address
                  </label>
                  <div
                    class="col-sm-8"
                  >
                    <input
                      class="form-control"
                      name="billingAddress"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="d-flex flex-row-reverse"
                >
                  <button
                    class="btn btn-primary py-2 px-3"
                    type="submit"
                  >
                    Submit
                  </button>
                  <button
                    class="btn btn-primary py-2 px-3 mr-2"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Users form UserManager Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="hero-wrap hero-wrap-2"
          data-stellar-background-ratio="0.5"
          style="background-image: url(images/bg_1.jpg);"
        >
          <div
            class="overlay"
          />
          <div
            class="container"
          >
            <div
              class="row no-gutters slider-text align-items-center justify-content-center"
            >
              <div
                class="col-md-9 text-center"
              >
                <h1
                  class="mb-2 bread"
                >
                  User Manager
                </h1>
                <p
                  class="breadcrumbs"
                >
                  <span
                    class="mr-2"
                  >
                    <a
                      href="/"
                    >
                      Home 
                      <i
                        class="ion-ios-arrow-forward"
                      />
                    </a>
                  </span>
                   
                  <span>
                    User Manager
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </span>
                </p>
              </div>
            </div>
          </div>
        </section>
        <section
          class="ftco-section"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-lg-10"
              >
                <div>
                  <p>
                    <em>
                      Loading...
                    </em>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
    <script
      defer=""
      nonce="undefined"
      src="https://accounts.google.com/gsi/client"
    />
  </body>,
  "container": <div>
    <div>
      <section
        class="hero-wrap hero-wrap-2"
        data-stellar-background-ratio="0.5"
        style="background-image: url(images/bg_1.jpg);"
      >
        <div
          class="overlay"
        />
        <div
          class="container"
        >
          <div
            class="row no-gutters slider-text align-items-center justify-content-center"
          >
            <div
              class="col-md-9 text-center"
            >
              <h1
                class="mb-2 bread"
              >
                User Manager
              </h1>
              <p
                class="breadcrumbs"
              >
                <span
                  class="mr-2"
                >
                  <a
                    href="/"
                  >
                    Home 
                    <i
                      class="ion-ios-arrow-forward"
                    />
                  </a>
                </span>
                 
                <span>
                  User Manager
                  <i
                    class="ion-ios-arrow-forward"
                  />
                </span>
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="ftco-section"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-10"
            >
              <div>
                <p>
                  <em>
                    Loading...
                  </em>
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
